components/core/src/clp/ffi/ir_stream/search/QueryHandler.hpp

```diff
@@ -0,0 +1,111 @@
+#ifndef CLP_FFI_IR_STREAM_SEARCH_QUERYHANDLER_HPP
+#define CLP_FFI_IR_STREAM_SEARCH_QUERYHANDLER_HPP
+
+#include <outcome/outcome.hpp>
+
+#include "../../KeyValuePairLogEvent.hpp"
+#include "../../SchemaTree.hpp"
+#include "AstEvaluationResult.hpp"
+#include "ErrorCode.hpp"
+#include "NewProjectedSchemaTreeNodeCallbackReq.hpp"
+
+namespace clp::ffi::ir_stream::search {
+/**
+ * Class for handing KV-pair IR stream search queries.
```

Comment:
```suggestion
 * Handler for KV-pair IR stream search queries.
```

Try to avoid "Class for" since it's usually redundant.